Release 2021-11-24

LoneRifle opened the pull request:
This release contains the following major changes:

- #832 
- #833
- #835
  - #834 
  - #848 
- #844 

It also contains the following:
- #831 
- #850

LoneRifle committed:
refactor(header): reorg layout, logo render

- Render the AskGov logo using the same scaffolding for agency logo,
  to keep DOM predictable
- Reorg layout for readability, specifying full device-level layout

wanlingt committed:
fix(image-upload): expand size of box for drag and drop file upload

re #746

wanlingt committed:
feat(image-edit): add resize and remove image buttons

re #746

LoneRifle committed:
build(deps-dev): downgrade lint-staged due to okonet/lint-staged#1056

LoneRifle committed:
chore: lint usePaginationRange, remove spurious extends

LoneRifle committed:
build(deps-dev): upgrade typescript-eslint

wanlingt committed:
Merge pull request #831 from opengovsg/build/deps/downgrade-lint-staged

build(deps-dev): rejig lint-staged, typescript-eslint

LinHuiqing committed:
feat(search): mount search query routes

LoneRifle committed:
refactor(header): extract styles to Chakra theme

Leave AuthLinks and WebsiteLinks style intact, as they are fairly
self-contained and can be more easily extracted out into their own

LoneRifle committed:
refactor: move remaining js(x) files to TypeScript

- Convert all remaining js/jsx files to TypeScript, adjusting
  for types in the process
- Remove references to React global, using proper imports instead
- Configure eslint to recognise the JSX global and ignore the
  react-scripts config overrides

LoneRifle committed:
Merge pull request #832 from opengovsg/refactor/header/chakra-theme

refactor(header): reorg logo render, migrate to Chakra theme

LoneRifle committed:
build(deps-dev): remove babel-eslint

LoneRifle committed:
Merge pull request #833 from opengovsg/refactor/jsx/begone

refactor: move remaining js(x) files to TypeScript

wanlingt committed:
refactor(image-theme): rename image themes to ImageBlock and ImageControl

wanlingt committed:
feat(image-edit): add edit alt text function

re #746

LinHuiqing committed:
fix(search): correct typo in env var SEARCH_PORT (#834)

LinHuiqing committed:
fix(search): modify routes to use relative path /

- includes relevant test changes on search.routes

dependabot[bot] committed:
build(deps-dev): bump lint-staged from 12.1.1 to 12.1.2 (#836)

Bumps [lint-staged](https://github.com/okonet/lint-staged) from 12.1.1 to 12.1.2.
- [Release notes](https://github.com/okonet/lint-staged/releases)
- [Commits](https://github.com/okonet/lint-staged/compare/v12.1.1...v12.1.2)

---
updated-dependencies:
- dependency-name: lint-staged
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump @aws-sdk/client-s3 from 3.41.0 to 3.42.0 in /server (#838)

Bumps [@aws-sdk/client-s3](https://github.com/aws/aws-sdk-js-v3/tree/HEAD/clients/client-s3) from 3.41.0 to 3.42.0.
- [Release notes](https://github.com/aws/aws-sdk-js-v3/releases)
- [Changelog](https://github.com/aws/aws-sdk-js-v3/blob/main/clients/client-s3/CHANGELOG.md)
- [Commits](https://github.com/aws/aws-sdk-js-v3/commits/v3.42.0/clients/client-s3)

---
updated-dependencies:
- dependency-name: "@aws-sdk/client-s3"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump neverthrow from 4.3.0 to 4.3.1 in /server (#837)

Bumps [neverthrow](https://github.com/supermacro/neverthrow) from 4.3.0 to 4.3.1.
- [Release notes](https://github.com/supermacro/neverthrow/releases)
- [Commits](https://github.com/supermacro/neverthrow/compare/v4.3.0...v4.3.1)

---
updated-dependencies:
- dependency-name: neverthrow
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump react-query from 3.33.2 to 3.33.4 in /client (#839)

Bumps [react-query](https://github.com/tannerlinsley/react-query) from 3.33.2 to 3.33.4.
- [Release notes](https://github.com/tannerlinsley/react-query/releases)
- [Commits](https://github.com/tannerlinsley/react-query/compare/v3.33.2...v3.33.4)

---
updated-dependencies:
- dependency-name: react-query
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/jest from 27.0.2 to 27.0.3 in /server (#840)

Bumps [@types/jest](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/jest) from 27.0.2 to 27.0.3.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/jest)

---
updated-dependencies:
- dependency-name: "@types/jest"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump lint-staged from 12.0.3 to 12.1.2 in /shared (#843)

Bumps [lint-staged](https://github.com/okonet/lint-staged) from 12.0.3 to 12.1.2.
- [Release notes](https://github.com/okonet/lint-staged/releases)
- [Commits](https://github.com/okonet/lint-staged/compare/v12.0.3...v12.1.2)

---
updated-dependencies:
- dependency-name: lint-staged
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump lint-staged from 12.0.3 to 12.1.2 in /server (#842)

Bumps [lint-staged](https://github.com/okonet/lint-staged) from 12.0.3 to 12.1.2.
- [Release notes](https://github.com/okonet/lint-staged/releases)
- [Commits](https://github.com/okonet/lint-staged/compare/v12.0.3...v12.1.2)

---
updated-dependencies:
- dependency-name: lint-staged
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump lint-staged from 12.0.3 to 12.1.2 in /client (#841)

Bumps [lint-staged](https://github.com/okonet/lint-staged) from 12.0.3 to 12.1.2.
- [Release notes](https://github.com/okonet/lint-staged/releases)
- [Commits](https://github.com/okonet/lint-staged/compare/v12.0.3...v12.1.2)

---
updated-dependencies:
- dependency-name: lint-staged
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

wanlingt committed:
fix(image-edit): remove edit alt text function

re #746

LinHuiqing committed:
Merge pull request #835 from opengovsg/feat/search/routes

feat(search): expose search query endpoint

LinHuiqing committed:
refactor(search): extract search client setup to a separate file (#848)

- for better modularity, based on `bootstrap/sequelize`

wanlingt committed:
fix(image-upload): add jpeg to accepted image upload formats

re #746

wanlingt committed:
Merge pull request #844 from opengovsg/feat/rte/edit-image

feat(rte): add resize and remove image options

dependabot[bot] committed:
build(deps-dev): bump eslint to 8.3.0 in /shared and /server (#822)

* build(deps-dev): bump eslint from 7.32.0 to 8.3.0 in /shared

Bumps [eslint](https://github.com/eslint/eslint) from 7.32.0 to 8.3.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.32.0...v8.3.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

* build(deps-dev): bump eslint from 7.32.0 to 8.3.0 in /server (#818)

Bumps [eslint](https://github.com/eslint/eslint) from 7.32.0 to 8.3.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.32.0...v8.3.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

LinHuiqing reviewed (commented):
Staging looks good to me :)

LoneRifle committed:
build(deps-dev): remove eslint + friends, lean on CRA instead (#850)

The client codebase explicitly specified eslint and several
configs/plugins, whose versions may conflict with what react-scripts
uses. Remove everything except prettier, leaning on react-scripts
as much as possible for our linting needs.

This goes against conventional wisdom to explicitly state your
dependencies, but should be fine given that the deps involved are used
for CLI or dev tooling, and should come from the batteries-included
react-scripts

dependabot[bot] committed:
build(deps): bump @testing-library/jest-dom in /client (#851)

Bumps [@testing-library/jest-dom](https://github.com/testing-library/jest-dom) from 5.15.0 to 5.15.1.
- [Release notes](https://github.com/testing-library/jest-dom/releases)
- [Changelog](https://github.com/testing-library/jest-dom/blob/main/CHANGELOG.md)
- [Commits](https://github.com/testing-library/jest-dom/compare/v5.15.0...v5.15.1)

---
updated-dependencies:
- dependency-name: "@testing-library/jest-dom"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
